app/views/drivers/show.html.erb

```diff
@@ -0,0 +1,79 @@
+<header>
+  <h1>Driver Details</h1>
+</header>
+<main>
+  <section class="show">
+      <div class="container-profile">
+        <div class="profile-pic">
+          <%= image_tag "driver_avatar.png", alt: "driver avatar", class: "driver" %>
+        </div>
+        <div class="profile">
+          <p><strong>Driver Name: </strong><%= @driver.name %></p>
+          <p><strong>Vin: </strong><%= @driver.vin %></p>
+          <p><strong>Availability Status:</strong><%= @driver.availability_status ? "AVAILABLE" : "UNAVAILABLE" %></p>
+          <p><strong>Total Earnings: </strong> $<%= '%.2f' %@driver.total_earnings %></p>
+          <p><strong>Average Rating: </strong> <%='%.1f' %@driver.average_rating %></p>
+          <div class="button-container">
+            <div>
+              <% if @driver.availability_status == true %>
+                <%= button_to "Change Availability", driver_path(@driver.id), method: :patch, params: {driver:{name: @driver.name, vin: @driver.vin, availability_status: false}}, class: "btn btn-secondary btn-sm" %>
+              <% else %>
+                <%= button_to "Change Availability", driver_path(@driver.id), method: :patch, params: {driver:{name: @driver.name, vin: @driver.vin, availability_status: true}}, class: "btn btn-secondary btn-sm" %>
+              <% end %>
```

Comment:
Consider refactoring using a variable and ternary to DRY up your code:


```suggestion
<% status  = @driver.availability_status == true ? false : true %> 
<%= button_to "Change Availability", driver_path(@driver.id), method: :patch, params: {driver:{name: @driver.name, vin: @driver.vin, availability_status: status}}, class: "btn btn-secondary btn-sm" %>
```

Reply:
On second thought, this is a lot of work for a view. Consider making a custom controller action that is `mark_available` or `mark_unavailable` and uses your custom model methods.